upgrade-cluster.yml

```diff
@@ -92,6 +92,7 @@
     - { role: kubernetes/client, tags: client }
     - { role: kubernetes/node-label, tags: node-label }
     - { role: kubernetes-apps/cluster_roles, tags: cluster-roles }
+    - { role: kubernetes-apps, tags: cinder-csi-driver }
```

Comment:
You say in the title that you want to install Cinder CSI before draining, but the draining happens in the pre-upgrade node. You mean that you want to have Cinder CSI installed before making the node reschedulable?

Reply:
You are correct, I'll update the PR to say "before first node is upgraded"

Reply:
Here we install specifically the Cinder CSI driver, because it is the only one in kubespray for now, but when we have other drivers (https://github.com/kubernetes-sigs/kubespray/pull/5549), maybe we should install all of them with a global tag csi-driver rather than having multiple tags.
But that's a thought for later, for now this seems appropriate.

Reply:
I have been thinking like that as well but didn't do anything now since there was only one supported when the code was written. However I agree that a tag named "csi driver" should be run and added to all future csi-providers

Reply:
I could fix a PR for streamlining CSI drivers later today using the tag "csi-driver" meaning this could also be used for the AWS driver